### PR TITLE
fix: memory leak caused by many unclosed redis subscriptions

### LINKED ISSUE #744: bug: memory leak with cleaning up subscriptions and tRPC observables
When an observable is closed and terminated, the callback for the subscription is never cleaned up and the subscription will remain.
Since new observable are created constantly upon each reload, we also create new redis subscriptions each time on the same clients.
Clients are limited to 10'000 on Redis and subscriptions are limited to 2^32 which is a limit that we are unlikely to reach but the memory and CPU usage due to the constant if statements being run could increase nevertheless.
Only workaround would be to restart Homarr entirely.
Memory leak size is unknown.

See https://github.com/homarr-labs/homarr/pull/743

### PR DESCRIPTION
**Pull request by Meierschlumpf**
<br/>
<div align="center">
  
  <h3>Homarr</h3>
</div>

**Thank you for your contribution. Please ensure that your pull request meets the following pull request:**

- [x] Builds without warnings or errors (``pnpm buid``, autofix with ``pnpm format:fix``)
- [x] Pull request targets ``dev`` branch
- [x] Commits follow the [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] No shorthand variable names are used (eg. ``x``, ``y``, ``i`` or any abbrevation)

Closes #744

### CODE REVIEW

File: packages/redis/src/lib/subscription-manager.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import { randomUUID } from "crypto";
+
+import type { MaybePromise } from "@homarr/common/types";
+import { logger } from "@homarr/log";
+
+import { createRedisConnection } from "./connection";
+
+type SubscriptionCallback = (message: string) => MaybePromise<void>;
+
+export class SubscriptionManager {
```
Comment (manuel-rw) [inline packages/redis/src/lib/subscription-manager.ts:10]: Add documentation what this does

Comment (manuel-rw) [inline packages/redis/src/lib/subscription-manager.ts:10]: ``Manager`` is a poorly chosen name

File: packages/redis/src/lib/subscription-manager.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    return () => {
+      logger.debug(`Removing redis channel callback channel='${channelName}' id='${id}'`);
+
+      const channelSubscriptions = this.subscriptions.get(channelName);
+      if (!channelSubscriptions) return;
+
+      channelSubscriptions.delete(id);
+
+      // If there are no subscriptions to the channel, unsubscribe from it
+      if (channelSubscriptions.size === 0) {
```
Comment (manuel-rw) [inline packages/redis/src/lib/subscription-manager.ts:54]: Invert to reduce nesting?

**Review by manuel-rw — CHANGES_REQUESTED** [review]

File: packages/redis/src/lib/subscription-manager.ts (hunk 1 above)
Comment (Meierschlumpf) [reply packages/redis/src/lib/subscription-manager.ts:10]: What do you think of `ChannelSubscriptionTracker`?
If not any better ideas?

Comment (manuel-rw) [reply packages/redis/src/lib/subscription-manager.ts]: lgtm 👍

**Review by manuel-rw — APPROVED** [review]

**Pull request merged**